TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs

```diff
@@ -89,13 +90,28 @@
             "/Persons/PersonDetail",
             model =>
             {
-                model.Filters.Add(new CheckPersonExistsFilter());
+                model.Filters.Add(new CheckPersonExistsFilterFactory());
             });
+
         options.Conventions.AddFolderApplicationModelConvention(
             "/Mqs/AddMq",
             model =>
             {
-                model.Filters.Add(new CheckPersonExistsFilter());
+                model.Filters.Add(new CheckPersonExistsFilterFactory());
+            });
+
+        options.Conventions.AddFolderApplicationModelConvention(
+            "/Mqs/DeleteMq",
+            model =>
+            {
+                model.Filters.Add(new ServiceFilterAttribute<CheckMandatoryQualificationExistsFilter>());
```

Comment:
is there a reason why this isn't done using the "factory" idea like CheckPersonExistsFilterFactory ?

Reply:
We can't use `ServiceFilterAttribute` for `CheckPersonExistsFilter` because of its `requireQts` parameter.

Reply:
ah